pkg/chunk/storage/factory.go

```diff
@@ -153,7 +153,10 @@ func NewStore(cfg Config, storeCfg chunk.StoreConfig, schemaCfg chunk.SchemaConf
 	stores := chunk.NewCompositeStore(cacheGenNumLoader)
 
 	for _, s := range schemaCfg.Configs {
-		index, err := NewIndexClient(s.IndexType, cfg, schemaCfg)
+		reg := prometheus.WrapRegistererWith(
```

Comment:
To avoid any misunderstanding with the outer `reg` variable, I would name it `indexReg`.